src/windows/mod.rs

```diff
@@ -0,0 +1,390 @@
+            if nt_headers.Signature == IMAGE_NT_SIGNATURE {
+                Some(nt_headers)
+            } else {
+                println!("NOT FOUND {:x}", nt_headers.Signature);
+                None
+            }
+        })
+    }
+
+    fn codeview_record70(&self) -> Option<&CodeViewRecord70> {
+        let bias = self.virtual_memory_bias().0;
+        unsafe {
+            let debug_dictionary: *const IMAGE_DEBUG_DIRECTORY =
+                mem::transmute(self.module_base().offset(bias));
+            if debug_dictionary.is_null() || (*debug_dictionary).Type != IMAGE_DEBUG_TYPE_CODEVIEW {
+                return None;
+            }
+            let debug_info: *const CodeViewRecord70 = mem::transmute(
+                self.module_base()
+                    .offset((*debug_dictionary).AddressOfRawData as isize),
+            );
+            if debug_info.is_null() || (*debug_info).signature != CV_SIGNATURE {
+                return None;
+            }
+            Some(&*debug_info)
+        }
+    }
+}
+
+impl<'a> SharedLibraryTrait for SharedLibrary<'a> {
+    type Segment = Segment<'a>;
+    type SegmentIter = SegmentIter<'a>;
+
+    #[inline]
+    fn name(&self) -> &OsStr {
+        &self.module_name
+    }
+
+    #[inline]
+    fn debug_name(&self) -> Option<&OsStr> {
+        self.codeview_record70().and_then(|codeview| unsafe {
+            let bytes: *const i8 = mem::transmute(&codeview.pdb_filename);
+            let cstr = CStr::from_ptr(bytes);
+            if let Ok(s) = cstr.to_str() {
+                Some(OsStr::new(s))
+            } else {
+                None
+            }
+        })
+    }
+
+    fn id(&self) -> Option<SharedLibraryId> {
+        self.nt_headers().map(|nt_headers| {
+            SharedLibraryId::PeSignature(
+                nt_headers.FileHeader.TimeDateStamp,
+                nt_headers.OptionalHeader.SizeOfImage,
+            )
+        })
+    }
+
+    #[inline]
+    fn debug_id(&self) -> Option<SharedLibraryId> {
+        self.codeview_record70().map(|codeview| unsafe {
+            SharedLibraryId::PdbSignature(mem::transmute(codeview.pdb_signature), codeview.pdb_age)
+        })
+    }
+
+    fn segments(&self) -> Self::SegmentIter {
+        let sections = self.nt_headers().map(|nt_headers| unsafe {
+            let base =
+                (nt_headers as *const _ as *const u8).add(mem::size_of::<IMAGE_NT_HEADERS>());
+            slice::from_raw_parts(
+                base as *const IMAGE_SECTION_HEADER,
+                nt_headers.FileHeader.NumberOfSections as usize,
+            )
+        });
+        SegmentIter {
+            sections: sections.unwrap_or(&[][..]),
+            phantom: PhantomData,
+        }
+    }
+
+    #[inline]
+    fn virtual_memory_bias(&self) -> Bias {
+        Bias(self.nt_headers().map_or(0, |nt_headers| {
+            nt_headers.OptionalHeader.DataDirectory[IMAGE_DIRECTORY_ENTRY_DEBUG as usize]
+                .VirtualAddress as isize
```

Comment:
This doesn't seem right to me. It's the RVA of the .debug section, whereas I would expect something like `module_base() - nt_headers.OptionalHeader.ImageBase`